src/invidious/helpers/youtube_api.cr

```diff
@@ -2,120 +2,398 @@
 # This file contains youtube API wrappers
 #
 
-# Hard-coded constants required by the API
-HARDCODED_API_KEY     = "AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8"
-HARDCODED_CLIENT_VERS = "2.20210330.08.00"
+module YoutubeAPI
+  extend self
 
-####################################################################
-# make_youtube_api_context(region)
-#
-# Return, as a Hash, the "context" data required to request the
-# youtube API endpoints.
-#
-def make_youtube_api_context(region : String | Nil) : Hash
-  return {
-    "client" => {
-      "hl"            => "en",
-      "gl"            => region || "US", # Can't be empty!
-      "clientName"    => "WEB",
-      "clientVersion" => HARDCODED_CLIENT_VERS,
+  # Enumerate used to select one of the clients supported by the API
+  enum ClientType
+    Web      = 0
+    Android  = 1
+    WebEmbed = 2
```

Comment:
Hmm. This is more of a question than a review but is defining a new client object for this even necessary? Imo we should only really be defining a new client when InnerTube requires something with a different client name or key. In the case of the embed, the only difference is the requirement of a `clientScreen`.

I'll say the "embed mode" should be enabled via config instead of being a new client option.

Reply:
As I said [in this comment](https://github.com/iv-org/invidious/pull/2277#discussion_r677400362), there is a different client name/version combo for the embedded player, hence that addition. Also, clients are not objects but constants, so in theory, they're just stored are an array of string pointers (nothing is allocated in memory).